### ISSUE TITLE
Preview fails on Windows because of path length in 2.5.0

### ISSUE THREAD

**Issue by Bananeweizen** [de-jcup/eclipse-asciidoctor-editor#414]
The change with SHA256 in the path completely breaks the preview on Windows for most input files. The path length of the temporary files goes beyond 254 in most cases, which is still a problem on Windows. In the following console output you can see the first 2 files failing, and only the 3rd one works because I created it in a workspace project with a short name and without nesting it in workspace folders.
```
ASP: Processing file:C:\dev\ascet\git\ascet\releng\com.etas.ascet.releng.architecture\documentation\architecture\_000_meta_data.adoc
Command convert_file failed.Failed:org.jruby.exceptions.SystemCallError: (ESRCH) No such process - asciidoctor: FAILED: <stdin>: Failed to load AsciiDoc document - No such process - C:\Users\kep9fe\AppData\Local\Temp\asciidoctor-editor-temp\project_com.etas.ascet.releng.architecture\ffd9ce1dcda52a6fae2c88040439c9dff749908986958bbdc576e2006f36fff6\36ab474035c3fe60dcfbfa37e41aaed07e2bbff114e408b0ae0696459a0ba8da_hidden-editorfile__000_meta_data.adoc
ASP: Processing file:C:\dev\ascet\git\ascet\releng\com.etas.ascet.releng.architecture\documentation\projectPage\_010_Scope.adoc
Command convert_file failed.Failed:org.jruby.exceptions.SystemCallError: (ESRCH) No such process - asciidoctor: FAILED: <stdin>: Failed to load AsciiDoc document - No such process - C:\Users\kep9fe\AppData\Local\Temp\asciidoctor-editor-temp\project_com.etas.ascet.releng.architecture\ffd9ce1dcda52a6fae2c88040439c9dff749908986958bbdc576e2006f36fff6\3d296a89c10161e7ac2c363a8058ec28644a750baaa46bc519ef720095e615c7_hidden-editorfile__010_Scope.adoc
ASP: Processing file:C:\dev\ascet\git\ascet-nightly\test.adoc
```
To be clear: The files _do_ exist on disk, they are just not accessible with that long path. For historic reasons, most Windows API comes in 2 variants (you may have seen the suffix A and W at function calls in Windows documentation). Only if all tools involved in a bigger process chain use the newer variant then paths longer than 254 characters can be used. And that seems not the case for the stack of Eclipse, Java, JRuby etc. here.

I really recommend to not create temporary paths and files on your own, but to simply use available Java methods like those from Apache Commons, Guava etc. See https://www.baeldung.com/java-temp-directories. Those would create _short_ unique names by just repeating the process in case of collision.

**Comment by DEV4c2** [comment]
Hi Bananeweizen, thanks for the explanation, we have the same problem here. In 2.4 it was no problem.
Can anyone help with a workaround or is it possible to revert to an older version?

**Comment by de-jcup** [comment]
Sorry for the problems on Windows. I have switch nearly 100% to Linux and was no longer aware about the 254 char problem. 

@DEV4c2 A quick workaround ... currently I do not know. But i try to deliver a new 2.5.1 version ASAP

**Comment by DEV4c2** [comment]
No Problem, thanx de-jcup! I just managed to reinstall 2.4.1 untill you could fix it.

**Comment by de-jcup** [comment]
With last commit I have changed the folder structure this way:
- Project folder 
  will be same name as eclipse project name. If there are duplicated projects we will use the same temp folder (so no unique identifier for projects any longer - which were introduced with #406 ). Destroying a project cache for same named projects will destroy project cache for both of them (but does really not matter, the next build/refresh will rebuild all necessary files)
- Filenames
  The filename uniqueness will no longer be handled by a SHA256 Key but instead by a simple UUID (but still using the path to the rendered file so the remapping from editor file to corresponding temp files can be done as before)
- Prefixes
  I switched from "hidden-editorfile_" to "hidden" only

So the pathes are much more shorter, but it is still possible to map easily back from an editor file to the corresponding temp file(s) without caching inside editor itself.

New temp file path looks now always:
`$systemTempFolder/$eclipseProjectName/$filePathAsUUID_$type_$filename.$fileEnding`

Here an example for an old windows path output (265 characters):
```
C:\Users\kep9fe\AppData\Local\Temp\asciidoctor-editor-temp\project_com.etas.ascet.releng.architecture\ffd9ce1dcda52a6fae2c88040439c9dff749908986958bbdc576e2006f36fff6\36ab474035c3fe60dcfbfa37e41aaed07e2bbff114e408b0ae0696459a0ba8da_hidden-editorfile_16_images.adoc
```

After the last commit same file will have a path like this (149 characters):
```
C:\Users\kep9fe\AppData\Local\Temp\asciidoctor-editor\all-testscripts-project$\1238d762-b42f-3329-8843-a7e5c4d740bb_hidden-editorfile_16_images.adoc
```
